### PR TITLE
Add Sphinx documentation

### PR DESCRIPTION
**Pull request by davehunt**
This addresses part of #7, the other part will be adding docstrings throughout the codebase.

The documentation based on this branch is available to view here:
http://pytest-selenium.readthedocs.org/en/latest/

Pinging @bobsilverberg for review.

### CODE REVIEW

**Comment by davehunt** [comment]
I think we can safely ignore the Travis CI errors for this pull request.

File: docs/installing.rst [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Installation
+============
+
+Requirements
+------------
+
+pytest-selenium requires Python 2.6 or 2.7.
```
Comment (bobsilverberg) [inline docs/installing.rst]: I thought you also designed it to support Python 3.

File: docs/installing.rst [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+Install pytest-selenium
```
Comment (bobsilverberg) [inline docs/installing.rst]: Would `Installing pytest-selenium` be a better section header?

File: docs/user_guide.rst [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+`documentation <https://github.com/SeleniumHQ/selenium/wiki/DesiredCapabilities#used-by-the-selenium-server-for-browser-selection>`_
+for details of accepted values. There are also a number of
+`browser specific capabilities <https://github.com/SeleniumHQ/selenium/wiki/DesiredCapabilities#browser-specific-capabilities>`_
+that can be set. Be sure to also check the documentation for your chosen
+driver, as the accepted capabilities may differ.
+
+Note that if your server is not running locally or is running on an alternate
+port you will need to specify the ``--host`` and ``--port`` command line
+options.
+
```
Comment (bobsilverberg) [inline docs/user_guide.rst]: You provide a sample command line for running with `Firefox`, `Chrome` and `PhantomJS`, but none when using `Remote`. I realize that there are more options and combinations for this, but perhaps a sample command showing specifying `--driver`, `--host` and `--port` would be helpful?

File: docs/user_guide.rst [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+Capabilities
+************
+
+Configuration options are specified using a capabilities dictionary. This is
+required when using a Selenium server to specify the target environment, but
+can also be used to configure local drivers.
+
+To specify capabilities, you can provide a JSON file on the command line using
+the `pytest-variables <https://github.com/davehunt/pytest-variables>`_ plugin.
+For example if you had a ```capabilties.json`` containing your capabilities, you
```
Comment (bobsilverberg) [inline docs/user_guide.rst]: One too many backticks before `capabilities.json`

File: docs/news.rst [hunk 5]
```diff
@@ -0,0 +1,5 @@
+Release Notes
+=============
+
+**0.1**
+* Initial release
```
Comment (bobsilverberg) [inline docs/news.rst]: These two lines appear in the output doc as `0.1 * Initial release`, where `0.1` is in bold. Is that what you intended? I think the asterisk after the `0.1` looks odd, as if it's pointing at a footnote.

**Comment by bobsilverberg** [comment]
This looks great, @davehunt. I have added a few comments above. I am not reviewing the Sphinx code, as I know nothing about it and don't understand it, but I assume you do, and it seems to be able to properly generate the docs.

One additional comment about `user_guide.rst `, while you have covered the different ways to run tests using `pytest-selenium`, I don't feel that you have completely covered how to use it. If all I had were these docs it would not be clear to me how to write a test that uses it. Do you think that should be added?

File: docs/user_guide.rst (hunk 3 above)
Comment (davehunt) [inline docs/user_guide.rst]: Good point, thanks.

File: docs/user_guide.rst (hunk 4 above)
Comment (davehunt) [inline docs/user_guide.rst]: +1 thanks

File: docs/installing.rst (hunk 1 above)
Comment (davehunt) [inline docs/installing.rst]: Not yet, there's #6 for that.

File: docs/news.rst (hunk 5 above)
Comment (davehunt) [inline docs/news.rst]: It should be a bullet, I was missing a blank line.

**Comment by davehunt** [comment]
> This looks great, @davehunt. I have added a few comments above. I am not reviewing the Sphinx code, as I know nothing about it and don't understand it, but I assume you do, and it seems to be able to properly generate the docs.

Sorry, I should have mentioned that the Sphinx code is boilerplate and not neccesary to review.

> One additional comment about `user_guide.rst `, while you have covered the different ways to run tests using `pytest-selenium`, I don't feel that you have completely covered how to use it. If all I had were these docs it would not be clear to me how to write a test that uses it. Do you think that should be added?

Absolutely! Would you be okay with this branch as it is and we can add a section on how to write your tests to take advantage of the plugin?

**Comment by bobsilverberg** [comment]
@davehunt Yes, this is good as far as it goes. r+

**Comment by bobsilverberg** [comment]
Oops. Hit the wrong button!

**Comment by davehunt** [comment]
I've added a quick start guide and a few more sections. When you're back could you take another look at this @bobsilverberg? Latest has been built and published here: http://pytest-selenium.readthedocs.org/en/latest/

**Comment by davehunt** [comment]
I've merged this to make it easier to keep up to date with master. I'm still keen for a review of the documentation, but any feedback can be fixed in new commits.

**Pull request closed without merging**